third_party/flux/cmd_manager.go

```diff
@@ -0,0 +1,212 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flux
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/vladimirvivien/gexe"
+	log "k8s.io/klog/v2"
+)
+
+type Opts struct {
+	name      string
+	source    string
+	namespace string
+	mode      string
+	url       string
+	branch    string
+	tag       string
+	commit    string
+	path      string
+	interval  string
+	args      []string
+}
+
+type Source string
+
+const (
+	Git    Source = "git"
+	Bucket Source = "bucket"
+	Helm   Source = "helm"
+	Oci    Source = "oci"
+)
+
+type Manager struct {
+	e          *gexe.Echo
+	kubeConfig string
+}
+
+type Option func(*Opts)
+
+func (m *Manager) processOpts(opts ...Option) *Opts {
+	option := &Opts{}
+	for _, op := range opts {
+		op(option)
+	}
+	return option
+}
+
+// WithNamespace is used to specify the namespace of flux installation
+func WithNamespace(namespace string) Option {
+	return func(opts *Opts) {
+		opts.namespace = namespace
+	}
+}
+
+// WithCommit is used to target a source with a specific commit SHA
+func WithCommit(commit string) Option {
+	return func(opts *Opts) {
+		opts.commit = commit
+	}
+}
+
+// WithTag is used to target a source with a specific tag
+func WithTag(tag string) Option {
+	return func(opts *Opts) {
+		opts.tag = tag
+	}
+}
+
+// WithBranch is used to target a source with a specific branch
+func WithBranch(branch string) Option {
+	return func(opts *Opts) {
+		opts.branch = branch
+	}
+}
+
+// WithPath is used to specify a path for reconciliation
+func WithPath(path string) Option {
+	return func(opts *Opts) {
+		opts.path = path
+	}
+}
+
+// WithInterval is used to specify how often flux should check for changes in a source
+func WithInterval(interval string) Option {
+	return func(opts *Opts) {
+		opts.interval = interval
+	}
+}
+
+// WithArgs is used to pass any additional parameter to Flux command
+func WithArgs(args ...string) Option {
+	return func(opts *Opts) {
+		opts.args = args
+	}
+}
+
+func (m *Manager) run(opts *Opts) (err error) {
+	if m.e.Prog().Avail("flux") == "" {
```

Comment:
Now that we have the template merged, this is more or less ready to be merged as well. 

Recently we enabled a way to use a custom path for these binaries that we use in the framework. i.e You can specify a path outside of the `$PATH` that you have your binaries in. Can we have a mechanism similar to that here? We have the same behavior on kind/and helm already. A way to customize teh path using `WithPath` kind of helper would be great to have.

Reply:
This is now fixed.